src/web/app/components/question-submission-form/question-submission-form.component.ts

```diff
@@ -26,7 +27,7 @@ import {
   templateUrl: './question-submission-form.component.html',
   styleUrls: ['./question-submission-form.component.scss'],
 })
-export class QuestionSubmissionFormComponent implements OnInit {
+export class QuestionSubmissionFormComponent implements DoCheck {
```

Comment:
Any reason for using `DoCheck` lifecycle here?

Reply:
When doing manual testing, it was observed that recipients were not sorted as desired (by the code below)
``` typescript
ngOnInit(): void {
    this.sortRecipientsByName();
  }
 ```
It was found out that this was due to lazy loading. The component has initialized however the recipient data is not yet loaded. Therefore, different lifecycle hook was thought to be necessary:
``` typescript
ngDoCheck(): void {
    if (!(this.isRecipientListSorted) && this.model.isLoaded) {
      this.isRecipientListSorted = true;
      this.sortRecipientsByName();
    }
  }
```
`ngDoCheck` was thought to be more suitable, as it would run after each call of `ngOnChanges()`, where the later function would be called every time Angular sets or resets data-bound input properties. (Which was assumed to be triggered after data is loaded)

Reply:
@wkurniawan07 Shall I create a issue and do a separate pull request for this?

Reply:
If it affects the production version, it should be a separate issue

Reply:
Understood, will do so.